core/src/main/java/com/sequenceiq/cloudbreak/conclusion/step/NetworkCheckerConclusionStep.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.cloudbreak.conclusion.step;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.cloudera.thunderhead.telemetry.nodestatus.NodeStatusProto.HealthStatus;
+import com.cloudera.thunderhead.telemetry.nodestatus.NodeStatusProto.NetworkDetails;
+import com.cloudera.thunderhead.telemetry.nodestatus.NodeStatusProto.NodeStatus;
+import com.cloudera.thunderhead.telemetry.nodestatus.NodeStatusProto.NodeStatusReport;
+import com.sequenceiq.cloudbreak.client.RPCResponse;
+import com.sequenceiq.cloudbreak.node.status.NodeStatusService;
+
+@Component
+public class NetworkCheckerConclusionStep extends ConclusionStep {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkCheckerConclusionStep.class);
+
+    @Inject
+    private NodeStatusService nodeStatusService;
+
+    @Override
+    public Conclusion check(Long resourceId) {
+        RPCResponse<NodeStatusReport> networkReport = nodeStatusService.getNetworkReport(resourceId);
+        if (networkReport.getResult() == null) {
+            String nodeStatusResult = networkReport.getFirstTextMessage();
+            LOGGER.info("Network report result was null, original message: {}", nodeStatusResult);
+            return succeeded();
+        }
+
+        List<String> networkFailures = new ArrayList<>();
+        List<String> networkFailureDetails = new ArrayList<>();
+        checkNetworkFailures(networkReport, networkFailures, networkFailureDetails);
+
+        if (networkFailures.isEmpty()) {
+            return succeeded();
+        } else {
+            return failed(networkFailures.toString(), networkFailureDetails.toString());
+        }
+    }
+
+    private void checkNetworkFailures(RPCResponse<NodeStatusReport> networkReport, List<String> networkFailures, List<String> networkFailureDetails) {
+        for (NodeStatus nodeStatus : networkReport.getResult().getNodesList()) {
+            NetworkDetails networkDetails = nodeStatus.getNetworkDetails();
+            String host = nodeStatus.getStatusDetails().getHost();
+            if (networkDetails.getCcmEnabled() && !HealthStatus.OK.equals(networkDetails.getCcmAccessible())) {
+                networkFailures.add("CCM is not accessible from node " + host + ". Please check network settings!");
+                String details = String.format("CCM health status is %s for node %s", networkDetails.getCcmAccessible(), host);
+                networkFailureDetails.add(details);
+                LOGGER.warn(details);
+            }
+            if (!HealthStatus.OK.equals(networkDetails.getClouderaComAccessible())) {
```

Comment:
note that UNKNOWN is the default status, that does not mean there is a problem with the connection. NOK could mean that so better to check against that.

Reply:
I will modify this.